### DISCUSSION TITLE
unable to load lib pi4j

### DISCUSSION THREAD

**Discussion by Desoriente695** [Pi4J/pi4j#176] [Q&A]
Hi,

We are beginners on raspberry pi and we want to use the pi4j library. The example program compiles but does not run :

GRAVE: Unable to load [libpi4j] using path: [/lib/raspberrypi/dynamic/libpi4j-armhf.so]

Someone has an idea to solve it ?

**Comment by FDelporte** [comment]
What version of Pi4J are you using? 
Can you share more information of how you build and run the application?

**Reply by Desoriente695** [reply]
We use Java 11 and we use pi4j 1.4

**Reply by Desoriente695** [reply]
We build the application with this command line : sudo pi4j --compile BlinkGpioExample.java

**Comment by FDelporte** [comment]
did you first install Pi4J with?
`curl -sSL https://pi4j.com/install | sudo bash`

for a Raspberry Pi 4 an update for WiringPi will also be required, see https://pi4j.com/1.4//install.html#Prerequisites

**Reply by Desoriente695** [reply]
yes we use this line (it is a rasp 3)

**Reply by Desoriente695** [reply]
We also updated pi4j

**Comment by Desoriente695** [comment]
You can find a screenshot of the error here :
https://drive.google.com/file/d/11vinBN6iP6kV4AyV26QYfmglysH61K4g/view?usp=sharing

**Reply by savageautomate** [reply]
> You can find a screenshot of the error here : https://drive.google.com/file/d/11vinBN6iP6kV4AyV26QYfmglysH61K4g/view?usp=sharing

Does your system have a /tmp folder? This is the location where Pi4J attempts to extract the native library files.

**Reply by Desoriente695** [reply]
yes there is a temp folder. But if the library is not install why it compil successfully ?

**Comment by Desoriente695** [comment]
compil : https://drive.google.com/file/d/1dxG67vg6pm9lC_kaoFXVRPX3d5codeQH/view?usp=sharing

pi4j version : https://drive.google.com/file/d/11UFx3ArSzPXt-_byFjjK_IuqVar-0hA8/view?usp=sharing

**Reply by micsix** [reply]
```
Looks like WiringPi may not be installed......should see something like this:

# pi4j --version

--------------------------------------------
            THE Pi4J PROJECT
--------------------------------------------
   PI4J.VERSION     : 1.4
   PI4J.TIMESTAMP   : 2021-03-02 03:43:12
  ----------------------------------------  
   WIRINGPI.PATH    : /usr/lib/libwiringPi.so /usr/local/lib/libwiringPi.so
   WIRINGPI.VERSION : 2.61
--------------------------------------------
```

**Reply by FDelporte** [reply]
there seems to be a mixup between V1.4 (which uses wiringpi) and V2 (uses pigpio).
